src/commons/components/Button/Button.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import styles from './Button.module.pcss';
+
+/**
+ * Button sizes
+ */
+export enum ButtonSize {
+  General = 'general',
+  Small = 'small'
+}
+
+/**
+ * Button styles
+ */
+export enum ButtonStyle {
+  Submit = 'submit',
+  Warning = 'warning',
+  Secondary = 'secondary',
+}
+
+interface Props extends React.ButtonHTMLAttributes<React.PropsWithChildren>{
+  /**
+   * Button style
+   */
+  styleType?: ButtonStyle | ButtonStyle.Submit,
+
+  /**
+   * Button size
+   */
+  buttonSize?: ButtonSize | ButtonSize.General,
+
+  /**
+   * Is button disabled
+   */
+  isDisabled?: boolean,
+
+  /**
+   * Is button loading
+   */
+  isLoading?: boolean,
+
+  /**
+   * Is button rounded
+   */
+  isRounded?: boolean,
+
+  /**
+   * Is button shacking
+   */
+  isShacking?: boolean,
+
+  /**
+   * Icon path for button
+   */
+  icon?: string,
+}
+
+/**
+ * Button component
+ *
+ * @param props - props of component
+ */
+const Button: React.FC<Props> = ({ ...props }) => {
```

Comment:
You can take props for styles. `{ disabled, loading, ...props}`. You don't need to write `props.` every time.